cobc/codegen.c

```diff
@@ -1830,14 +1830,31 @@ output_gnucobol_defines (const char *formatted_date)
 	output_line ("#define  COB_MODULE_TIME\t\t%d", i);
 
 	{
-		struct cb_text_list *l = cb_include_file_list ;
-		for (;l;l=l->next){
-			if (l->text[0] == '<'){
+		struct cb_text_list *l = cb_include_file_list;
+		struct cb_text_list *ld = cb_include_file_list_directive;
+		struct cb_text_list *next;
+		for (;l;l=l->next) {
+			if (l->text[0] == '<') {
 				output_line ("#include %s", l->text);
 			} else {
 				output_line ("#include \"%s\"", l->text);
 			}
 		}
+		while (ld) {
+			if (ld->text[0] == '<') {
+				output_line ("#include %s", ld->text);
+			} else {
+				output_line ("#include \"%s\"", ld->text);
+			}
+			next = ld->next;
+			if (ld != NULL) {
+				cobc_free (ld);
+				cobc_free (ld->text);
+				ld = NULL;
+			}
+			ld = next;
+		}
```

Comment:
the code is nearly good (we cannot free ld first, then access its element to free that), but please change it to have no `ld`:

```c
		struct cb_text_list *l;
		for (l = cb_include_file_list; l; l = l->next) {
		  ...
		}
		l = cb_include_file_list_directive;
		while (l) {
		  struct cb_text_list last = l;		
		  ...
		  l = l->next:
		  cobc_free (last->text);
		  cobc_free (last);
		}
```